src/blob_gc_job.cc

```diff
@@ -149,21 +149,25 @@ Status BlobGCJob::SampleCandidateFiles() {
   std::vector<BlobFileMeta*> result;
   for (const auto& file : blob_gc_->inputs()) {
     bool selected = false;
-    Status s = DoSample(file, &selected);
+    uint64_t estimate_discardable_size = 0;
+    Status s = DoSample(file, &selected, estimate_discardable_size);
     if (!s.ok()) {
       return s;
     }
     if (selected) {
       result.push_back(file);
-    }
+    }else{
+      // this update the discardable size of this file according to sample
+      file->AddDiscardableSize(estimate_discardable_size);
```

Comment:
why? IMO `AddDiscardableSize` is to collect discardable size from table properties of deleted SSTs, which is orthogonal to the result of sampling.

Reply:
because discardable_size of recovered file is 0 after a re-open so I think maybe I can set an estimated discardable size by the result of sample  if this file is not picked by gc_picker......seems there maybe some mistake, only the file just recovered need this step......or there maybe some other better way(I just think maybe we can use the result of sample)

Reply:
Even so, the estimation from sampling is not compatible with discardable size updated from table property. I think we should not mix the two, otherwise the value of discardable size would be hard to reason about.